code/game/objects/effects/mainttraps.dm

```diff
@@ -0,0 +1,356 @@
+#define PICK_STYLE_RANDOM 1
+#define PICK_STYLE_ORDERED 2
+#define PICK_STYLE_ALL 3
+
+/obj/effect/trap
+	name = "nasty trap"
+	desc = "doesn't do much, really..."
+	icon = 'icons/obj/items_and_weapons.dmi'
+	icon_state = "gavelblock"
+	invisibility = INVISIBILITY_OBSERVER
+	var/reusable = FALSE //can it trigger more than once
+	var/inuse = FALSE //used to make sure it dont get used when it shouldnt
+
+/obj/effect/trap/proc/TrapEffect(AM)
+	return TRUE
+
+/obj/effect/trap/trigger //this only triggers other traps
+	name = "pressure pad"
+	desc = "seems flimsy"
+	var/grounded = FALSE //does it ignore fliers
+	var/pick_style = PICK_STYLE_ORDERED
+	var/requirehuman = TRUE
+
+/obj/effect/trap/trigger/Crossed(AM as mob|obj)
+	if(isturf(loc))
+		if(ismob(AM) && grounded)
+			var/mob/MM = AM
+			if(!(MM.movement_type & FLYING))
+				if(TrapEffect(AM))
+					if(!reusable)
+						qdel(src)
+		else if(!requirehuman)
+			if(TrapEffect(AM))
+				if(!reusable)
+					qdel(src)
+		else if(ismob(AM))
+			var/mob/MM = AM
+			if(MM.mind)
+				if(TrapEffect(AM))
+					if(!reusable)
+						qdel(src)
+
+/obj/effect/trap/trigger/TrapEffect(AM)
+	if(inuse)
+		return FALSE
+	else 
+		inuse = TRUE
+	var/list/possibletraps = list()
+	for(var/obj/effect/trap/nexus/payload in oview(10, src))
+		possibletraps += payload
```

Comment:
This list should probably just be populated once in LateInitialize rather than every time. `oview()` is expensive.